iceoryx_posh/include/iceoryx_posh/mepoo/chunk_settings.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_MEPOO_CHUNK_SETTINGS_HPP
+#define IOX_POSH_MEPOO_CHUNK_SETTINGS_HPP
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_utils/cxx/expected.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace mepoo
+{
+class ChunkSettings
+{
+  public:
+    enum class Error
```

Comment:
Nitpick: Other classes using `expected` define this outside of the class and call it `ChunkSettingsError`

Reply:
yes, but I wanted to test the approach with the nested class and I find it better since its more clear and one change the name of the class without forgetting to change the name of the enum. I can change it if it bothers you much.

Reply:
public nested class is preferable since it ties the error to the ChunkSettings..